anesthetic/samples.py

```diff
@@ -114,25 +115,35 @@ def plot(self, ax, paramname_x, paramname_y=None, *args, **kwargs):
             Pandas array of axes objects
 
         """
-        plot_type = kwargs.pop('plot_type', 'kde')
+        plot_type = kwargs.pop('plot_type', {'1d': 'kde', '2d': 'kde'})
```

Comment:
Currently this code expects ``types`` to be a dictionary, which breaks backward compatibility, since previously all code expects a string (hence the failing tests). You should keep the same default as before, but have a line something along the lines of:
```python
if isinstance(plot_type, str):
    plot_type = {'1d':plot_type, '2d':plot_type}
```
This will allow it to handle both cases

Reply:
Do you mean `types` or `plot_type`, here? Or both?

Reply:
Sorry, I meant ``types``, i.e.
```python
if isinstance(types, str):
    types = {'1d':types, '2d':types}
```

Reply:
I'm getting confused, my apologies:

 In ``MCMCSamples.plot_2d`` (where there is no reference to ``types``) you should have the code:

```python
if isinstance(plot_type, str):
    plot_type = {'1d':plot_type, '2d':plot_type}
```

whilst in ``make_2d_axes`` the ``types`` keyword should be removed

Reply:
Do you mean `MCMCSamples.plot()`? Because `plot_2d` does reference `types`. `plot` indeed doesn't and I've reverted things there.

Reply:
Indeed you are correct. I've been looking at a wrong version of the code.

On master, ``MCMCSamples.plot_2d`` expects a keyword argument ``types`` which at the moment can be either a 
- str (in which case it produces a triangle plot),
- list (in which case it produces a rectangle plot with two different types above and below the diagonal)

One way to implement the diagonal behaviour we want would be to allow this to take a length three list (or a dictionary) indicating the upper, lower and diagonal behaviour.   I think that a dictionary would be a lot nicer than how it's currently implemented. We could make it backward compatible by converting lists appropriately, but that may not be essential if the new functionality is a lot clearer.

``MCMCSamples.plot`` should remain unchanged in it's processing of plot_type, i.e. it should always take a string, but now you can pass it ``'hist'`` for the case that it's a 1d plot (and it should throw an ``NotImplementedError`` if you try to do a 2d histogram plot).

Reply:
* reverted `MCMCSamples.plot` (see recent commit)
* changed `types` to a dictionary with keywords `'diagonal'`, `'lower'` and `'upper'` and updated docs accordingly. Should be backwards compatible with `str` and `list` though.

Should I add a deprecation warning that now the use of a dictionary is advised?

Reply:
That would be great

Reply:
done, I made it a FutureWarning as that is what pops up in jupyter notebooks, too.